ApplicationInsights/JavaScriptSDK/Telemetry/PageViewManager.ts

```diff
@@ -44,74 +47,82 @@ export class PageViewManager {
         }
 
-        var pageViewSent = false;
-        var customDuration = undefined;
-
-        if (PageViewPerformance.isPerformanceTimingSupported()) {
-            var start = PageViewPerformance.getPerformanceTiming().navigationStart;
-            customDuration = PageViewPerformance.getDuration(start, +new Date);
-
-            if (!PageViewPerformance.shouldCollectDuration(customDuration)) {
-                customDuration = undefined;
-            }
-        } else {
+        // if performance timing is not supported by the browser, send the page view telemetry with the duration provided by the user. If the user
+        // do not provide the duration, set duration to undefined
+        if (!PageViewPerformance.isPerformanceTimingSupported()) {
             this.appInsights.sendPageViewInternal(
-                name,
-                url,
-                !isNaN(duration) ? duration : undefined,
-                properties,
-                measurements);
+                pageView,
+                customProperties);
             this._channel.flush();
-            pageViewSent = true;
-        }
 
-        if (!pageViewSent && (this.overridePageViewDuration || !isNaN(duration))) {
-            // 1, 2, 4 cases
-            this.appInsights.sendPageViewInternal(
-                name,
-                url,
-                !isNaN(duration) ? duration : customDuration,
-                properties,
-                measurements);
-                this._channel.flush();
-            pageViewSent = true;
-        }
-
-        var maxDurationLimit = 60000;
-
-        if (!PageViewPerformance.isPerformanceTimingSupported()) {
             // no navigation timing (IE 8, iOS Safari 8.4, Opera Mini 8 - see http://caniuse.com/#feat=nav-timing)
             _InternalLogging.throwInternal(
                 LoggingSeverity.WARNING,
                 _InternalMessageId.NavigationTimingNotSupported,
                 "trackPageView: navigation timing API used for calculation of page duration is not supported in this browser. This page view will be collected without duration and timing info.");
+
             return;
         }
 
+        var pageViewSent = false;
+        var customDuration = undefined;
+
+        // if the performance timing is supported by the browser, calculate the custom duration
+        var start = PageViewPerformance.getPerformanceTiming().navigationStart;
+        customDuration = PageViewPerformance.getDuration(start, +new Date);
+        if (!PageViewPerformance.shouldCollectDuration(customDuration)) {
+            customDuration = undefined;
+        }
+
+        // if the user has provided duration, send a page view telemetry with the provided duration. Otherwise, if
+        // overridePageViewDuration is set to true, send a page view telemetry with the custom duration calculated earlier
+        let duration = pageView.duration
+        if (this.overridePageViewDuration || !isNaN(duration)) {
```

Comment:
(!pageViewSent &&

Reply:
discussed with Piyali offline.

---
In reply to: [205510324](https://github.com/Microsoft/ApplicationInsights-JS/pull/635#discussion_r205510324) [](ancestors = 205510324)